templates/skeleton/app/components/Header.tsx

```diff
@@ -159,15 +163,19 @@ function CartToggle({cart}: Pick<HeaderProps, 'cart'>) {
   return (
     <Suspense fallback={<CartBadge count={null} />}>
       <Await resolve={cart}>
-        {(cart) => {
-          if (!cart) return <CartBadge count={0} />;
-          return <CartBadge count={cart.totalQuantity || 0} />;
-        }}
+        <CartBanner />
       </Await>
     </Suspense>
   );
 }
 
+function CartBanner() {
+  const originalCart = useAsyncValue() as CartApiQueryFragment | null;
+  const cart = useOptimisticCart(originalCart);
+  const linesCount = cart.totalQuantity ?? null;
+  return <CartBadge count={linesCount} />;
```

Comment:
Or something like

```suggestion
  return <CartBadge count={cart?.totalQuantity ?? 0} />;
```

Reply:
This sounds better, but would like to keep it `?? null` as it's the same behavior as suspense fallback.